management/src/main/java/io/micronaut/management/endpoint/env/EnvironmentEndpoint.java

```diff
@@ -17,104 +17,116 @@
 
 import io.micronaut.context.env.Environment;
 import io.micronaut.context.env.PropertySource;
+import io.micronaut.core.annotation.Nullable;
 import io.micronaut.management.endpoint.annotation.Endpoint;
 import io.micronaut.management.endpoint.annotation.Read;
 import io.micronaut.management.endpoint.annotation.Selector;
 
+import java.security.Principal;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collection;
 import java.util.Comparator;
 import java.util.LinkedHashMap;
-import java.util.List;
 import java.util.Map;
-import java.util.regex.Pattern;
-import java.util.stream.Collectors;
 
 /**
  * {@link Endpoint} that displays information about the environment and its property sources.
  *
  * @author Álvaro Sánchez-Mariscal
  * @since 1.2.0
  */
-@Endpoint(EnvironmentEndpoint.NAME)
+@Endpoint(
+        id = EnvironmentEndpoint.NAME,
+        defaultEnabled = EnvironmentEndpoint.DEFAULT_ENABLED
+)
 public class EnvironmentEndpoint {
 
     /**
      * Endpoint name.
      */
     public static final String NAME = "env";
 
-    public static final String[] PROPERTY_NAMES_TO_MASK = new String[] {
-            "password", "credential", "certificate", "key", "secret", "token"
-    };
+    /**
+     * Endpoint default enabled.
+     */
+    public static final boolean DEFAULT_ENABLED = false;
 
     private final Environment environment;
-    private final List<Pattern> maskPatterns;
+    private final EnvironmentEndpointFilter environmentFilter;
 
     /**
      * @param environment The {@link Environment}
+     * @param environmentFilter The registered {@link EnvironmentEndpointFilter} bean if one is registered
      */
-    public EnvironmentEndpoint(Environment environment) {
+    public EnvironmentEndpoint(Environment environment, @Nullable EnvironmentEndpointFilter environmentFilter) {
```

Comment:
Need to add a new constructor and not modify this one to avoid breaking binary compatiblity

Reply:
I tried with both, and the `EnvironmentEndpointSpec` tests failed (I guess because the one arg constructor got used even though an `EnvironmentEndpointFilter` was defined as a bean)

Is there a way to get it to prefer the two-arg constructor?  Would `@Inject` help?  I'll see

Reply:
I think `@Inject` does it! 😎  [`7bc7553` (#6711)](https://github.com/micronaut-projects/micronaut-core/pull/6711/commits/7bc7553cef4cd65382f892ae950339dcd7042495)